### PR TITLE
Add movies guide

### PR DESCRIPTION
**Pull request by AlexicaWright**

### CODE REVIEW

File: documentation/movies.workspace.adoc [hunk 1]
```diff
@@ -0,0 +1,731 @@
...
+
+== Find nodes
+
+To start, can you find the actor Tom Hanks?
+
+Run the following query:
+
+[source,cypher]
+----
+MATCH (tom {name: "Tom Hanks"})
```
Comment (jexp) [inline documentation/movies.workspace.adoc:560]: use `:Person` label here

Comment (jexp) [reply documentation/movies.workspace.adoc:560]: change tom to `p` or `person`

File: documentation/movies.workspace.adoc [hunk 2]
```diff
@@ -0,0 +1,731 @@
...
+----
+
+Next, can you find a movie called _Cloud Atlas_?
+
+[%collapsible]
+.Reveal the solution
+====
+[source,cypher]
+----
+MATCH (movie {title:"Cloud Atlas"})
```
Comment (jexp) [inline documentation/movies.workspace.adoc:571]: add `:Movie` label

File: documentation/movies.workspace.adoc [hunk 3]
```diff
@@ -0,0 +1,731 @@
...
+[source,cypher]
+----
+MATCH (people:Person)
+RETURN people.name LIMIT 10
+----
+
+[TIP]
+====
+Try to remove the property key `.name` from the RETURN clause and see the difference.
+(`RETURN people LIMIT 10`)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:587]: use the full statement here

File: documentation/movies.workspace.adoc [hunk 4]
```diff
@@ -0,0 +1,731 @@
...
+
+[source,cypher]
+----
+MATCH (tom {name: "Tom Hanks"})
+RETURN tom
+----
+
+Next, can you find a movie called _Cloud Atlas_?
+
+[%collapsible]
```
Comment (jexp) [inline documentation/movies.workspace.adoc:566]: Nice! more of those :)

File: documentation/movies.workspace.adoc [hunk 5]
```diff
@@ -0,0 +1,731 @@
...
+====
+Try to remove the property key `.name` from the RETURN clause and see the difference.
+(`RETURN people LIMIT 10`)
+====
+
+If you want to find movies released in a specific timespan, you can add conditions by using `WHERE`:
+
+[source,cypher]
+----
+MATCH (nineties:Movie) WHERE nineties.released >= 1990 AND nineties.released < 2000
```
Comment (jexp) [inline documentation/movies.workspace.adoc:594]: change `nineties` to `movie`

File: documentation/movies.workspace.adoc [hunk 6]
```diff
@@ -0,0 +1,731 @@
...
+In the next step you will look for patterns in the graph.
+
+== Find patterns
+
+A pattern in the graph is a specific arrangement of nodes and relationships that can be matched in the graph.
+To find which movies Tom Hanks have acted in, specify the pattern as follows:
+
+[source,cypher]
+----
+MATCH (tom:Person {name: "Tom Hanks"})-[:ACTED_IN]->(tomHanksMovies)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:607]: change `tom` to `actor` or `person` and `tomHanksMovies` to `movie`

File: documentation/movies.workspace.adoc [hunk 7]
```diff
@@ -0,0 +1,731 @@
...
+
+Can you specify a pattern to find who directed _Cloud Atlas_?
+Hint: The relationship between a person who is a director and the movie they directed is `:DIRECTED`.
+
+[%collapsible]
+.Reveal the solution
+====
+[source,cypher]
+----
+MATCH (cloudAtlas {title: "Cloud Atlas"})<-[:DIRECTED]-(directors)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:619]: Add `:Movie` label change `cloudAtlas` to `movie`

File: documentation/movies.workspace.adoc [hunk 8]
```diff
@@ -0,0 +1,731 @@
...
+RETURN directors.name
+----
+====
+
+If you extend the length of the pattern, you can find co-actors:
+
+[source,cypher]
+----
+MATCH (tom:Person {name:"Tom Hanks"})-[:ACTED_IN]->(m)<-[:ACTED_IN]-(coActors)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:628]: change `tom` to `person` or better leave it off

File: documentation/movies.workspace.adoc [hunk 9]
```diff
@@ -0,0 +1,731 @@
...
+RETURN coActors.name
+----
+
+You can return _all_ properties if you are not sure exactly what you are looking for:
+
+[source,cypher]
+----
+MATCH (people:Person)-[relatedTo]-(:Movie {title: "Cloud Atlas"})
+RETURN people.name, Type(relatedTo), relatedTo
```
Comment (jexp) [inline documentation/movies.workspace.adoc:637]: lowcase `type()`

File: documentation/movies.workspace.adoc [hunk 10]
```diff
@@ -0,0 +1,731 @@
...
+----
+MATCH (tom:Person {name:"Tom Hanks"})-[:ACTED_IN]->(m)<-[:ACTED_IN]-(coActors)
+RETURN coActors.name
+----
+
+You can return _all_ properties if you are not sure exactly what you are looking for:
+
+[source,cypher]
+----
+MATCH (people:Person)-[relatedTo]-(:Movie {title: "Cloud Atlas"})
```
Comment (jexp) [inline documentation/movies.workspace.adoc:636]: change `people` to `person` we should be consistent in singular for variables

File: documentation/movies.workspace.adoc (hunk 7 above)
Comment (jexp) [reply documentation/movies.workspace.adoc:619]: change `directors` to  `director`

File: documentation/movies.workspace.adoc [hunk 11]
```diff
@@ -0,0 +1,731 @@
...
+MATCH (movie {title:"Cloud Atlas"})
+RETURN movie
+----
+====
+
+Similarly, you can find all the people (nodes with the `Person` label) in the graph, but limit the returned number to ten:
+
+[source,cypher]
+----
+MATCH (people:Person)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:580]: change `people` to `person`

File: documentation/movies.workspace.adoc [hunk 12]
```diff
@@ -0,0 +1,731 @@
...
+== Six degrees of Kevin Bacon
+
+You may have heard about the concept that any two people on Earth are six or fewer connections apart.
+In Hollywood, this is said to be true for Kevin Bacon and any other actor.
+
+The following query lets you see all movies and actors up to six hops away from Kevin Bacon:
+
+[source,cypher]
+----
+MATCH (bacon:Person {name:"Kevin Bacon"})-[*1..6]-(hollywood)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:653]: `bacon` -> leave off, we don't use it in the query
`hollywood` -> `person`

File: documentation/movies.workspace.adoc [hunk 13]
```diff
@@ -0,0 +1,731 @@
...
+----
+
+But is it true that every actor is really connected to Kevin Bacon?
+
+Cypher has a built-in algorithm for this, `ShortestPath` and it looks like this:
+
+[source,cypher]
+----
+MATCH p=shortestPath(
+(bacon:Person {name:"Kevin Bacon"})-[*]-(meg:Person {name:"Meg Ryan"})
```
Comment (jexp) [inline documentation/movies.workspace.adoc:664]: leave off `bacon` and `meg`

Comment (jexp) [reply documentation/movies.workspace.adoc:664]: change `p` to `path`

File: documentation/movies.workspace.adoc [hunk 14]
```diff
@@ -0,0 +1,731 @@
...
+
+A basic recommendation approach is to find connections past an immediate neighborhood of nodes which are themselves well-connected.
+For Tom Hanks, this means to find actors who he has yet to work with, but who has worked with his previous co-actors.
+Then you need to find someone who can introduce Tom Hanks to his potential new co-actor.
+
+To find a list of potential co-actors, run the following query:
+
+[source,cypher]
+----
+MATCH (tom:Person {name:"Tom Hanks"})-[:ACTED_IN]->(m)<-[:ACTED_IN]-(coActors),
```
Comment (jexp) [inline documentation/movies.workspace.adoc:683]: change `tom` to `person` or `actor` in all places

File: documentation/movies.workspace.adoc [hunk 15]
```diff
@@ -0,0 +1,731 @@
...
+----
+
+The `WHERE NOT` part of the query ensures that you don't get actors who already worked with Tom Hanks nor Tom Hanks himself.
+The results are ordered by the number of co-co-actors they have in common.
+
+Now, if you think it would be a good idea for Tom Hanks and Keanu Reeves to do a movie together, who would be able to introduce them?
+
+[source,cypher]
+----
+MATCH (tom:Person {name:"Tom Hanks"})-[:ACTED_IN]->(m)<-[:ACTED_IN]-(coActors),
```
Comment (jexp) [inline documentation/movies.workspace.adoc:696]: change `tom` to `person` or `actor`

File: documentation/movies.workspace.adoc [hunk 16]
```diff
@@ -0,0 +1,731 @@
...
+  (coActors)-[:ACTED_IN]->(m2)<-[:ACTED_IN]-(keanu:Person {name:"Keanu Reeves"})
+RETURN tom, m, coActors, m2, keanu
+----
+
+In the last step, you are going to clean up your database by deleting the data you have created.
+
+== Clean up
+
+This dataset is fun to experiment with, but once you are done, you can delete everything with the following:
```
Comment (jexp) [inline documentation/movies.workspace.adoc:705]: make `*delete everything*` bold.

File: documentation/movies.workspace.adoc [hunk 17]
```diff
@@ -0,0 +1,731 @@
...
+
+[source,cypher]
+----
+MATCH (n)
+DETACH DELETE n
+----
+
+Note that this does in fact permanently delete the nodes and relationships in this database, so use with caution.
```
Comment (jexp) [inline documentation/movies.workspace.adoc:713]: WARNING:

File: documentation/movies.workspace.adoc [hunk 18]
```diff
@@ -0,0 +1,731 @@
...
+You can verify that there is nothing left by a simple query:
+
+[source,cypher]
+----
+MATCH (n)
```
Comment (jexp) [inline documentation/movies.workspace.adoc:718]: Probably better: `MATCH () RETURN count(*)`

File: documentation/movies.workspace.adoc [hunk 19]
```diff
@@ -0,0 +1,731 @@
...
+RETURN n
+----
+
+Well done!
+
+If you want to learn more about Cypher, see the following:
+
+* https://graphacademy.neo4j.com/categories/cypher/[Cypher Online Courses^]
+* https://neo4j.com/docs/cypher-manual/current/introduction/[Cypher Manual^]
+* https://neo4j.com/docs/cypher-cheat-sheet/5/auradb-enterprise/[Cypher Cheat-Sheet^].
```
Comment (jexp) [inline documentation/movies.workspace.adoc:728]: https://neo4j.com/docs/cypher-cheat-sheet/ should resolve automatically to current

File: documentation/movies.workspace.adoc [hunk 20]
```diff
@@ -0,0 +1,731 @@
...
+
+You are going to create a dataset with `Person` and `Movie` nodes and different types of relationships to connect them.
+
+image::movie-model.png[]
+
+Use the command below but be careful to only run it **once** as it will add data every time it is run.
+
+[source,cypher]
+----
+CREATE (TheMatrix:Movie {title:'The Matrix', released:1999, tagline:'Welcome to the Real World'})
```
Comment (jexp) [inline documentation/movies.workspace.adoc:40]: Should we switch the CREATE statements to MERGE in the setup, so running it multiple times will not mess it up?

File: documentation/movies.workspace.adoc [hunk 21]
```diff
@@ -0,0 +1,731 @@
...
+CREATE (Keanu:Person {name:'Keanu Reeves', born:1964})
+CREATE (Carrie:Person {name:'Carrie-Anne Moss', born:1967})
+CREATE (Laurence:Person {name:'Laurence Fishburne', born:1961})
+CREATE (Hugo:Person {name:'Hugo Weaving', born:1960})
+CREATE (LillyW:Person {name:'Lilly Wachowski', born:1967})
+CREATE (LanaW:Person {name:'Lana Wachowski', born:1965})
+CREATE (JoelS:Person {name:'Joel Silver', born:1952})
+CREATE
+(Keanu)-[:ACTED_IN {roles:['Neo']}]->(TheMatrix),
```
Comment (jexp) [inline documentation/movies.workspace.adoc:49]: for MERGE these need to be separate lines, no comma

**Review by jexp — APPROVED** [review]
LGTM thanks a lot! Also tested all the statements.

File: documentation/movies.workspace.adoc (hunk 2 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: ✅

File: documentation/movies.workspace.adoc (hunk 5 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: Changed it to `m` instead for consistency.

File: documentation/movies.workspace.adoc (hunk 6 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: ✅

File: documentation/movies.workspace.adoc (hunk 8 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: changed it to `p` for consistency

File: documentation/movies.workspace.adoc (hunk 9 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: ✅

File: documentation/movies.workspace.adoc (hunk 10 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: `p` as above.

File: documentation/movies.workspace.adoc (hunk 11 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: As above ✅

File: documentation/movies.workspace.adoc (hunk 12 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: ✅

File: documentation/movies.workspace.adoc (hunk 13 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: ✅

File: documentation/movies.workspace.adoc [hunk 22]
```diff
@@ -0,0 +1,731 @@
...
+Then you need to find someone who can introduce Tom Hanks to his potential new co-actor.
+
+To find a list of potential co-actors, run the following query:
+
+[source,cypher]
+----
+MATCH (tom:Person {name:"Tom Hanks"})-[:ACTED_IN]->(m)<-[:ACTED_IN]-(coActors),
+  (coActors)-[:ACTED_IN]->(m2)<-[:ACTED_IN]-(cocoActors)
+WHERE NOT (tom)-[:ACTED_IN]->()<-[:ACTED_IN]-(cocoActors) AND tom <> cocoActors
+RETURN cocoActors.name AS Recommended, count(*) AS Strength ORDER BY Strength DESC
```
[the remark this answers is not in the document]
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: ✅

File: documentation/movies.workspace.adoc (hunk 16 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc:689]: ✅

File: documentation/movies.workspace.adoc (hunk 20 above)
Comment (AlexicaWright) [reply documentation/movies.workspace.adoc]: It would be convenient but as we are talking about creating data, it makes more sense to use `CREATE`, IMHO.

**Pull request merged**